src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,116 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private User user;
+    private RegistrationServiceImpl registrationService;
+
+    @AfterEach
+    void tearDown() {
+        registrationService.clearStorage();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setId(1L);
+        user.setLogin("loginOne");
+        user.setAge(20);
+        user.setPassword("werewjdskfj");
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void userValidParameters_Ok() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void userAlreadyPresent_NotOk() {
+        registrationService.register(user);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void userLoginNull_NotOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void userPasswordNull_NotOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void userAgeNull_NotOk() {
+        user.setAge(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void userAgeLessZero_NotOk() {
+        user.setAge(-56);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void userAgeNotAdult_NotOk() {
+        user.setAge(5);
```

Comment:
краще брати гранично недопустимі значення 
```suggestion
        user.setAge(17);
```